extensions/xcode/src/services/xcode-simulator-application.service.ts

```diff
@@ -315,4 +337,98 @@ export class XcodeSimulatorApplicationService {
       (simulatorSandBoxPath) => simulatorSandBoxPath.bundleIdentifier === bundleIdentifier
     )?.sandBoxPath;
   }
+
+  /**
+   * Find AppGroup directory path for a given XcodeSimulator and bundle identifier
+   * @param simulator The XcodeSimulator
+   * @param bundleIdentifier The bundle identifier
+   * @private
+   */
+  private static async findAppGroupDirectoryPath(
```

Comment:
As this function is a code duplication (with tiny adjustments) of the `findSandboxDirectoryPath` it would be better to either write a more generic function which can find the sandbox as well as the app group or any other directory path or a function which retrieves the sandbox and app group directory path at once.